library/ViewPager.js

```diff
@@ -187,7 +187,7 @@ export default class ViewPager extends Component {
         //A trick to solve bugs on Android. Delay a little
         setTimeout(this.scrollToPage.bind(this, this.props.initialPage, true), 0);
       }
-    } else if (this.layoutChanged) {
+    } else if (this.layoutChanged || prevProps.pageDataArray.length !== this.props.pageDataArray.length) {
```

Comment:
this is useful and necessary when pageDataArray changed , for instance , you deleted one photo!